scripts/new_project.kts

```diff
@@ -9,23 +9,67 @@ object NewProject {
     private const val TEMPLATE_FOLDER_NAME = "CoroutineTemplate"
     private const val TEMPLATE_PACKAGE_NAME = "co.nimblehq.coroutine"
 
+    private const val APP_PATTERN = "^[A-Z\$][a-zA-Z0-9(\\s)\$]*\$"
+    private const val PACKAGE_PATTERN = "^[a-z]+(\\.[a-z][a-z0-9]*)*\$"
+
     private val modules = listOf("app", "data", "domain")
 
     private var appName = ""
-    private val appNameWithoutSpace: String
-        get() = appName.replace(" ", "")
     private var packageName = ""
+
     private val projectPath: String
-        get() = rootPath + appNameWithoutSpace
+        get() = rootPath + appName.replace(oldValue = " ", newValue = "")
+
     private val rootPath: String
         get() = System.getProperty("user.dir").replace("scripts", "")
 
     fun generate(args: Array<String>) {
         handleArguments(args)
-        initializeNewProjectFolder()
-        cleanNewProjectFolder()
-        renamePackageNameFolders()
-        renamePackageNameWithinFiles()
+    }
+
+    private fun handleArguments(args: Array<String>) {
+        val agrumentError = "ERROR: Invalid Agrument name: Ensure define argruments => app-name={\"MyProject\"} or {\"My Project\"} { package-name={com.sample.myproject}"
+        when (args.size) {
+            1 -> when {
+                args.first().startsWith(KEY_APP_NAME) -> showMessage("ERROR: No package has been provided")
+                args.first().startsWith(KEY_PACKAGE_NAME) -> showMessage("ERROR: No app name has been provided")
+                else -> showMessage(agrumentError)
+            }
+            2 -> args.map { arg ->
+                val (key, value) = arg.split(ARGUMENT_DELIMITER)
+                when (key) {
+                    KEY_APP_NAME -> validAppName(value)
+                    KEY_PACKAGE_NAME -> validPackageName(value)
+                    else -> showMessage(agrumentError)
+                }.also { executeNextSteps() }
+            }
+            else -> showMessage("ERROR: Require app-name and package-name to initialize the new project")
+        }
+    }
```

Comment:
@Wadeewee to simplify the condition and avoid nested conditions, I think we can show the error if `app-name` or `package-name` is not provided, no need for a specific check for each case. What do you think? cc @Tuubz @kaungkhantsoe 
Something like:
```suggestion
    private fun handleArguments(args: Array<String>) {
        args.forEach {
              val (key, value) = it.split(ARGUMENT_DELIMITER)
              when (key) {
                    KEY_APP_NAME -> appName = value
                    KEY_PACKAGE_NAME -> packageName = value
              }
        }
        if (appName.isEmpty() or packageName.isEmpty()) {
            showMessage("ERROR: Invalid Agrument name: Ensure define argruments => app-name={\"MyProject\"} or {\"My Project\"} { package-name={com.sample.myproject}")
            // exit process
        }
        validateAppName(appName)
        validatePackageName(packageName)
    }
```

Reply:
@Wadeewee I think we should exit the process when there is an error, so there is only 1 error that will display 😁  What do you think?
@Tuubz I know that this implementation is slightly different from the AC, but is there any chance to update the AC as I mention above? 🙇 `no need for a specific check for each case`

Reply:
@manh-t I agree with you Manh, if you scroll up I also suggested a solution where we exit the process ☝️ 

Anyways, if it's different from the AC, I'm fine with that 👍